Formula/newrelic-infra-agent.rb

```diff
@@ -0,0 +1,40 @@
+class NewrelicInfraAgent < Formula
+  desc "New Relic infrastructure agent"
+  homepage "https://github.com/newrelic/infrastructure-agent"
+  url "https://github.com/newrelic/infrastructure-agent/archive/refs/tags/1.20.0.tar.gz"
+  sha256 "b9f6f744b8376b0ba92657237902d8a8e7c12cbb7f56c6f13f93b52255f31224"
+  license "Apache-2.0"
+  head "https://github.com/newrelic/infrastructure-agent.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["VERSION"] = version
+    ENV["GOOS"] = "darwin"
+    ENV["GOARCH"] = "amd64"
+    ENV["CGO_ENABLED"] = "1"
+    system "make", "dist-for-os"
+    on_macos do
+      bin.install "dist/darwin-newrelic-infra_darwin_amd64/newrelic-infra"
+      bin.install "dist/darwin-newrelic-infra-ctl_darwin_amd64/newrelic-infra-ctl"
+      bin.install "dist/darwin-newrelic-infra-service_darwin_amd64/newrelic-infra-service"
```

Comment:
```suggestion
    goarch = Hardware::CPU.arm? ? "arm64" : "amd64"
    ENV["VERSION"] = version
    ENV["GOOS"] = "darwin"
    ENV["GOARCH"] = goarch
    ENV["CGO_ENABLED"] = "1"
    system "make", "dist-for-os"
    on_macos do
      bin.install "dist/darwin-newrelic-infra_darwin_#{goarch}/newrelic-infra"
      bin.install "dist/darwin-newrelic-infra-ctl_darwin_#{goarch}/newrelic-infra-ctl"
      bin.install "dist/darwin-newrelic-infra-service_darwin_#{goarch}/newrelic-infra-service"
```
We don't want to set `GOARCH=amd64` on ARM.

Reply:
Is it possible to bind this formula to darwin/amd64 only for now? We have m1 darwin on the roadmap and this is an MVP. For linux we have completely different flow, so we are not interested in linux version of brew now.

Reply:
> be maintained (i.e. the last release wasn’t ages ago, it works without patching on all Homebrew-supported OS versions and has no outstanding, unpatched security vulnerabilities)

https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

Reply:
We should get this to build on M1 and Linux. If we don't, this formula will make it on another Homebrew maintainer's to-do list.

I don't want to add a formula that I know will create work for another maintainer.

Reply:
Just in case, do you know any way to test M1 mac (any cloud tested already) if you don't own physical one?

Reply:
AWS has M1 nodes, though they're expensive. Scaleway is a bit more reasonable.

Do you think this won't build on M1? Because if all you're compiling is Go code, then I don't think you have anything to worry about.

Reply:
Our sponsor macstadium has some: https://www.macstadium.com/m1-mini

Reply:
> AWS has M1 nodes, though they're expensive. Scaleway is a bit more reasonable.

We checked AWS, there is only intel based bare metal options. Could you check which region did you find them? 
Another option (https://www.macstadium.com/m1-mini) will be complicated as we need to follow internal process of validation for each product we use and it can take months. 😞

> Do you think this won't build on M1? Because if all you're compiling is Go code, then I don't think you have anything to worry about.

For the m1 compilation, we need to validate that CGO dependent libraries will work as expected. 

For the linux, we will put some effort to bring this to the formula, likely we have amd64/arm64/arm fully tested there.

Reply:
Ah, you're right. I was misinformed about AWS. My apologies.